packages/ui/components/topbanner/TopBanner.tsx

```diff
@@ -0,0 +1,48 @@
+import { ExclamationIcon } from "@heroicons/react/outline";
+import { XIcon } from "@heroicons/react/solid";
+import classNames from "classnames";
+import { ReactNode, useState } from "react";
+
+export type TopBannerProps = {
+  text: string;
+  variant?: keyof typeof variantClassName;
+  actions?: ReactNode;
+};
+
+const variantClassName = {
+  default: "bg-gradient-primary",
+  warning: "bg-orange-400",
+  error: "bg-red-400",
+};
+
+export function TopBanner(props: TopBannerProps) {
+  const { variant = "default", text } = props;
+  const [isOpen, setOpen] = useState(true);
+
+  return (
+    <div
+      id="banner"
+      className={classNames(
+        " z-50 flex  h-[40px] w-full items-start justify-between gap-8 border border-b border-gray-200 bg-gray-50 px-4 text-center sm:items-center",
+        variantClassName[variant],
+        isOpen ? "sticky" : "hidden"
+      )}>
+      <div className="flex flex-1 items-center justify-center gap-2">
+        <p className="flex items-center justify-center gap-2 font-sans text-sm font-medium leading-4 text-gray-900">
+          {variant === ("warning" || "error") && (
```

Comment:
Does this actually work? IIRC this will never reach error.

```suggestion
          {["warning", "error"].includes(variant) && (
```